internal/e2e/e2eutil/compare_documents.go

```diff
@@ -99,18 +107,98 @@ func getFile(script *testscript.TestScript, filePath string) *os.File {
 	return file
 }
 
-func compareMetaData(script *testscript.TestScript, first, second *sbom.Metadata) bool {
-	firstStr := first.String()
-	secondStr := second.String()
+func compareMetaData(script *testscript.TestScript, have, want *sbom.Metadata) bool {
+	equal := true
+
+	if have.GetId() != want.GetId() {
```

Comment:
Probably want to use a switch case here, so it doesn't need to check every single condition. The first time it hits false, it breaks out

```go
switch {
case have.GetId() != want.GetId():
    // log and return `false`
...
}
```

Reply:
If implementing the solution from my other comment, this comment should be ignored